Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,330 @@
+            window.minSize = new Vector2(380.0f, 680.0f);
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            LoadIcons();
+
+            string[] modeStrings = { "Record", "Playback" };
+            Mode = (ToolMode)GUILayout.SelectionGrid((int)Mode, modeStrings, modeStrings.Length);
+
+            switch (mode)
+            {
+                case ToolMode.Recording:
+                    DrawRecordingGUI();
+                    break;
+                case ToolMode.Playback:
+                    DrawPlaybackGUI();
+                    break;
+            }
+
+            EditorGUILayout.Space();
+
+// XXX Reloading the scene is currently not supported,
+// due to the life cycle of the MRTK "instance" object.
+// Enable the button below once scene reloading is supported!
+#if false
+            using (new GUIEnabledWrapper(Application.isPlaying))
+            {
+                bool reloadScene = GUILayout.Button("Reload Scene");
+                if (reloadScene)
+                {
+                    Scene activeScene = SceneManager.GetActiveScene();
+                    if (activeScene.IsValid())
+                    {
+                        SceneManager.LoadScene(activeScene.name);
+                        return;
+                    }
+                }
+            }
+#endif
+        }
+
+        private void DrawRecordingGUI()
+        {
+            if (!Application.isPlaying)
+            {
+                EditorGUILayout.HelpBox("Input test recording is only available in play mode", MessageType.Info);
+                return;
+            }
+            if (RecordingService == null)
+            {
+                EditorGUILayout.HelpBox("No input recording service found", MessageType.Info);
+                return;
+            }
+
+            using (new GUILayout.HorizontalScope())
+            {
+                bool newUseTimeLimit = GUILayout.Toggle(RecordingService.UseBufferTimeLimit, "Use buffer time limit");
+                if (newUseTimeLimit != RecordingService.UseBufferTimeLimit)
+                {
+                    RecordingService.UseBufferTimeLimit = newUseTimeLimit;
+                }
+
+                using (new GUIEnabledWrapper(RecordingService.UseBufferTimeLimit))
+                {
+                    float newTimeLimit = EditorGUILayout.FloatField(RecordingService.RecordingBufferTimeLimit);
+                    if (newTimeLimit != RecordingService.RecordingBufferTimeLimit)
+                    {
+                        RecordingService.RecordingBufferTimeLimit = newTimeLimit;
+                    }
+                }
+            }
+
+            bool wasRecording = RecordingService.IsRecording;
+            bool record = GUILayout.Toggle(wasRecording, wasRecording ? new GUIContent(iconRecordActive, "Stop recording input animation") : new GUIContent(iconRecord, "Record new input animation"), "Button");
```

Comment:
Super nit: Another fairly long line - could be helpful to bring the wasRecording ? thing down to the next line at least.

Reply:
bool record = GUILayout.Toggle(
                           wasRecording,
                           wasRecording ?
                               new GUIContent(iconRecordActive, "Stop recording input animation") :
                               new GUIContent(iconRecord, "Record new input animation"),
                           "Button");

Reply:
damnit it doesn't preserve my spaces nooooo